eq-author/src/App/QuestionnairesPage/QuestionnairesView/index.test.js

```diff
@@ -811,5 +810,114 @@ describe("QuestionnairesView", () => {
         expect(getByText("1 of 2")).toBeTruthy();
       });
     });
+    describe("Filtering", () => {
+      it("should filter unowned questionnaires out when Owner mode selected", () => {
+        props.questionnaires = [
+          buildQuestionnaire(1, {
+            createdBy: {
+              id: "3",
+              name: "test user",
+              email: "test@email.com",
+              displayName: "test user",
+            },
+          }),
+          buildQuestionnaire(2, {
+            createdBy: {
+              id: "2",
+              name: "test user",
+              email: "test@email.com",
+              displayName: "test user",
+            },
+          }),
+          buildQuestionnaire(3, {
+            createdBy: {
+              id: "1",
+              name: "test user",
+              email: "test@email.com",
+              displayName: "test user",
+            },
+          }),
+        ];
+
+        const { getByLabelText, queryByText } = render(
+          <QuestionnairesView
+            {...props}
+            questionnaires={props.questionnaires}
+          />
+        );
+
+        const showAllButton = getByLabelText("All");
+        const hideUnownedButton = getByLabelText("Owner");
+
+        expect(queryByText("Questionnaire 1 Title")).toBeTruthy();
+        expect(queryByText("Questionnaire 2 Title")).toBeFalsy();
+        expect(queryByText("Questionnaire 3 Title")).toBeFalsy();
```

Comment:
Are these 3 lines necessary as you are already asserting that the button works on the last 3 lines? If you want to assert that the initial state is sorting by owner, maybe move it to another test, so it is more clear what you are testing?

Reply:
These three lines are just to show what we expect the starting state to be, and the 2nd group asseerts we can toggle too the new mode, and the 3rd group asserts we can toggle back.

Reply:
I understand that, but will it make more sense to move the first group to its own test where we are testing that the initial state should be Owned?